src/classes/HH_AutoCompleteDataProvider_LCTRL.cls

```diff
@@ -53,6 +53,7 @@ public with sharing class HH_AutoCompleteDataProvider_LCTRL {
             String preparedQueryValue = '%' + queryValue.replaceAll('([%_\\\\])', '\\\\$0') + '%';
             
             // use helper to get all contact fields to support custom household naming and greetings.
+            // note that our helper does FLS, so we don't have to do it here.
```

Comment:
I find it very confusing to use property getters this way.  You're using the conSelect property like a function call.. a function call that takes implicit arguments from global scope, rather than being passed in, even.

I urge you to consider changing the conSelect property to a method call with a descriptive name.  Maybe even something like `getContactSelectQueryWithFls()`?